### PR TITLE
Store created + updated date as part of VaccineData files (also includes ability to regen vaccine data files)

### PR DESCRIPTION
**Pull request by theramis**
## Changes
It's best to review this PR by commits IMO. Also probably turn on hide whitespace for differences. 

1. Created a concept of `DataDocument` which can store a generic object into the DataStore. It adds to the generic object some metadata (created and updated date)
2. Updated VaccineDataStore to use the `DataDocument` 
3. Unfortunately due to change #2 it meant reading the current json files doesn't work since the datastructure had changed. Rather than manually updating the files, I made fetchy have the ability to regenerate all of the files. This will be useful as we decide to extract more stuff from the page as we can regenerate our data structure from the raw site again. 

## TODO
- ~~Due to the regen, it deletes the data files which don't have rawSites at the moment. Wait until https://github.com/theramis/aucvid/pull/133 is merged and then regen the files for this PR.~~ **DONE**

### CODE REVIEW

File: fetchy/fetchyboi.ts [hunk 1]
```diff
@@ -65,4 +82,12 @@ const main = async () => {
   }
 };
 
+const main = async () => {
+  if (process.env.REGEN_DATA === "true") {
```
Comment (andycarrell) [inline fetchy/fetchyboi.ts:86]: Can we possibly be really explicit with this variable?

```suggestion
  if (process.env.REGENERATE_DATA === "true") {
```

File: fetchy/fetchyboi.ts [hunk 2]
```diff
@@ -29,34 +31,49 @@ const sendDataUpdatedNotification = async () => {
   );
 };
 
-const areObjectsDifferent = (a: object, b: object) =>
-  Object.keys(diff(a, b)).length !== 0;
+const areObjectsDifferent = (a: Object, b: Object) =>
```
Comment (andycarrell) [inline fetchy/fetchyboi.ts:34]: I think actually lowercase object is possibly better here - I sure don't know haha
```suggestion
const areObjectsDifferent = (a: object, b: object) =>
```

File: app/propsGenerators/indexPagePropsGenerator.ts [hunk 3]
```diff
@@ -23,10 +23,12 @@ export default async function fetchIndexPageProps(): Promise<IndexPageProps> {
...
-      yesterdayData.vaccinationsPerDhb
+      latestData.data.vaccinationsPerDhb,
+      yesterdayData.data.vaccinationsPerDhb
     ),
-    dataValidAsAtTimeUtc: DateTime.fromISO(latestData.dataValidAsAtNzTimeIso)
+    dataValidAsAtTimeUtc: DateTime.fromISO(
+      latestData.data.dataValidAsAtNzTimeIso
+    )
       .toUTC()
       .toISO(),
```
Comment (andycarrell) [inline app/propsGenerators/indexPagePropsGenerator.ts:33]: Should this just be stored in UTC? Not part of this PR but something to consider 🙏

**Comment by andycarrell** [comment]
PR looks good - if it were me, I'd create a completely separate function for regeneration - I don't really understand why we'd need to Regen based on config like that?

File: data/vaccineData/2021-10-29.json [hunk 4]
```diff
@@ -1,215 +1,221 @@
...
-      "firstDosesPercentage": 0.12,
-      "firstDosesTo90Percent": 13416,
-      "secondDoses": 1653,
-      "secondDosesPercentage": 0.1,
-      "secondDosesTo90Percent": 13803,
-      "totalPopulation": 17173
-    }
-  ],
-  "dataValidAsAtNzTimeIso": "2021-10-29T23:59:00.000+13:00"
+  "metaData": {
```
Comment (jishaal) [inline data/vaccineData/2021-10-29.json:2]: Minor but `metadata` should be lowercase

**Review by jishaal — COMMENTED** [review]
LGTM. Cool to have the regen functionality now 👍🏼

**Review by jishaal — APPROVED** [review]

File: fetchy/fetchyboi.ts (hunk 1 above)
Comment (theramis) [reply fetchy/fetchyboi.ts:86]: sweet will do agreed.

File: fetchy/fetchyboi.ts (hunk 2 above)
Comment (theramis) [reply fetchy/fetchyboi.ts:34]: I have no idea haha, I think your commit made them upper case so I left it like that 😂

File: app/propsGenerators/indexPagePropsGenerator.ts (hunk 3 above)
Comment (theramis) [reply app/propsGenerators/indexPagePropsGenerator.ts:33]: yea agree, will do some other time in a diff PR

**Comment by theramis** [comment]
> PR looks good - if it were me, I'd create a completely separate function for regeneration - I don't really understand why we'd need to Regen based on config like that?

TBH same, I got lazy honestly

**Review by andycarrell — APPROVED** [review]
